flycheck.el

```diff
@@ -6920,6 +6921,62 @@ See URL `http://www.erlang.org/'."
    (error line-start (file-name) ":" line ": " (message) line-end))
   :modes erlang-mode)
 
+(defun contains-rebar-config (dir-name)
+  "Return DIR-NAME if DIR-NAME/rebar.config exists, nil otherwise."
+  (if (file-exists-p (expand-file-name "rebar.config" dir-name))
+      dir-name nil))
+
+(defun locate-rebar3-project-root (file-name &optional prev-file-name acc)
```

Comment:
To be clearer, any Erlang application in a project could have a `rebar.config` of its own, while there will also be a top level `rebar.config`. So this just makes sure that the file we are using isn't a sub-app or dependency of the project.